raft/raft.go

```diff
@@ -434,6 +448,7 @@ func (r *raft) reset(term uint64) {
 		}
 	}
 	r.pendingConf = false
+	r.readOnly = newReadOnly(r.readOnly.option)
```

Comment:
Don't we need to do something here to tell the application that any previous reads have failed and will need to be retried?